_shared/project/pyproject.toml

```diff
@@ -24,11 +24,13 @@ filterwarnings = [
 
 {% if cookiecutter.get("linter") == "ruff" %}
 [tool.ruff]
-target-version = "py311"
+target-version = "py{{ python_versions()|oldest|pyformat(PyFormats.MAJOR_MINOR_FMT) }}"
```

Comment:
Don't hard-code the Python version. Notice that this sets Ruff's target version to the _oldest_ Python version that the project supports, which is the right thing to do (we don't want `make lint` telling us to use Python features that we can't use yet).